### PR TITLE
[2.7] bpo-30450: Pull Windows dependencies from GitHub rather than SVN (GH-1783)

### PR DESCRIPTION
**Pull request by zware**
This also contains (partial) cherry-picks of other PRs since then.

https://bugs.python.org/issue30450

### CODE REVIEW

File: Doc/make.bat [hunk 1]
```diff
@@ -5,8 +5,19 @@ pushd %~dp0
...
+call ..\PCBuild\find_python.bat %PYTHON%
+if "%SPHINXBUILD%" EQU "" if "%PYTHON%" NEQ "" (
+    set SPHINXBUILD=%PYTHON%\..\Scripts\sphinx-build.exe
+    rem Cannot use %SPHINXBUILD% in the same block where we set it
+    if not exist "%PYTHON%\..\Scripts\sphinx-build.exe" (
+        echo Installing sphinx with %PYTHON%
+        "%PYTHON%" -m pip install sphinx
+        if errorlevel 1 exit /B
+    )
+)
```
Comment (zooba) [inline Doc/make.bat]: I have an open PR making this bit _even better_! You should review that before backporting something different :)

File: PCbuild/python.props [hunk 2]
```diff
@@ -32,9 +32,9 @@
 
     <!-- Directories of external projects. tcltk is handled in tcltk.props -->
     <ExternalsDir>$([System.IO.Path]::GetFullPath(`$(PySourcePath)externals\`))</ExternalsDir>
-    <sqlite3Dir>$(ExternalsDir)sqlite-3.8.11.0\</sqlite3Dir>
+    <sqlite3Dir>$(ExternalsDir)sqlite-3.14.2.0\</sqlite3Dir>
```
Comment (zooba) [inline PCbuild/python.props:35]: Not sure if this is a change we wanted to actually make...

**Review by zooba — APPROVED** [review]
Looks good. Thanks for grabbing the Doc/make.bat change

File: PCbuild/python.props (hunk 2 above)
Comment (zware) [reply PCbuild/python.props:35]: This was simpler than creating a new branch for an old version of sqlite, and @benjaminp basically approved it on IRC (you never replied :)).

**Comment by zware** [comment]
@zooba Have I missed any other PRs that affect these scripts?  I just found that last one via `git log master .github/appveyor.yml`.

**Comment by zooba** [comment]
Not yet, though I'm working on one to enable the v141 toolset and latest Win10 SDKs. Not really needed for 2.7

**Comment by zware** [comment]
Ok, I'm not concerned about that for 2.7.  Once CI is happy with the latest commit here, I'll go ahead and merge.

**Pull request merged**